src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/Table.java

```diff
@@ -320,14 +321,26 @@ public void addRow(T item, List<Content> contents) {
             }
         }
         int colIndex = 0;
+        Pattern pattern = Pattern.compile("<a\\b");
         for (Content c : contents) {
             HtmlStyle cellStyle = columnStyles.get(colIndex);
+            // Only make element tabbable if it doesn't contain tabbable content to avoid widget_tabbable_single(Potential Violation)
+            // For more info refer : https://able.ibm.com/rules/archives/latest/doc/en-US/widget_tabbable_single.html
+            boolean matchFound = pattern.matcher(c.toString()).find();
             // Always add content to make sure the cell isn't dropped
             var cell = HtmlTree.DIV(cellStyle).addUnchecked(c.isEmpty() ? Text.EMPTY : c);
             cell.addStyle(rowStyle);
+            if (!matchFound) {
+                cell.put(HtmlAttr.ROLE, "row")
+                    .put(HtmlAttr.TABINDEX, "0");
+            }
 
             for (String tabClass : tabClasses) {
                 cell.addStyle(tabClass);
+                if (!matchFound) {
+                    cell.put(HtmlAttr.ROLE, "row")
```

Comment:
We now use [CSS Grid layout](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_grid_layout) for tables. Unfortunately, this implies that we do not have HTML elements representing rows. Tables are represented by an unstructured stream of table cells. This means that [`role="cell"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/cell_role) would probably be the right attribute to use here.

Once [Subgrid](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout/Subgrid) is well enough supported in browsers (which may be soon), we will be able to introduce elements representing table rows.

Reply:
The problem I see is that we do not currently have elements representing table rows. There is [a JBS issue](https://bugs.openjdk.org/browse/JDK-8288900) for this problem, and it looks like we might be able to solve it in the not too distant future, depending on browsers supporting the necessary CSS features.  However, until we do have actual elements representing table rows, I can't see how it makes sense to use `role="row"` attributes on what is essentially a table cell, not a row.